addons/mail/models/mail_thread.py

```diff
@@ -4598,3 +4598,23 @@ def get_views(self, views, options=None):
         if "form" in res["views"] and isinstance(self.env[self._name], self.env.registry['mail.activity.mixin']):
             res["models"][self._name]["has_activities"] = True
         return res
+
+    @api.model
+    def _get_thread_with_access(self, thread_model, thread_id, token, _hash, pid):
+        thread = self.with_context(active_test=False).search([("id", "=", thread_id)])
+        return thread.with_context(active_test=True)
+
+    def _prepare_post_data(self, post_data, special_mentions, **kwargs):
+        self.ensure_one()
+        if "body" in post_data:
+            post_data["body"] = Markup(post_data["body"])  # contains HTML such as @mentions
```

Comment:
This code belongs to controller, not model, as it is specific to the controller itself after checking security

Reply:
As explained on another comment, ok for me too to move this one.